metadata-ingestion/src/datahub/ingestion/source/unity/proxy.py

```diff
@@ -88,72 +102,40 @@ def assigned_metastore(self) -> Metastore:
         response = self._workspace_client.metastores.summary()
         return self._create_metastore(response)
 
-    def metastores(self) -> Iterable[Metastore]:
-        response = self._workspace_client.metastores.list()
-        if not response:
-            logger.info("Metastores not found")
-            return []
-        for metastore in response:
-            yield self._create_metastore(metastore)
-
     def catalogs(self, metastore: Metastore) -> Iterable[Catalog]:
-        response: dict = self._unity_catalog_api.list_catalogs()
-        num_catalogs: int = 0
-        if response.get("catalogs") is None:
-            logger.info(f"Catalogs not found for metastore {metastore.name}")
+        response = self._workspace_client.catalogs.list()
+        if not response:
+            logger.info("Catalogs not found")
             return []
-
-        for obj in response["catalogs"]:
-            if obj["metastore_id"] == metastore.metastore_id:
-                yield self._create_catalog(metastore, obj)
-                num_catalogs += 1
-
-        if num_catalogs == 0:
-            logger.info(
-                f"Catalogs not found for metastore where metastore_id is {metastore.metastore_id}"
-            )
+        for catalog in response:
+            yield self._create_catalog(metastore, catalog)
 
     def schemas(self, catalog: Catalog) -> Iterable[Schema]:
-        response: dict = self._unity_catalog_api.list_schemas(
-            catalog_name=catalog.name, name_pattern=None
-        )
-        if response.get("schemas") is None:
-            logger.info(f"Schemas not found for catalog {catalog.name}")
+        response = self._workspace_client.schemas.list(catalog_name=catalog.name)
+        if not response:
+            logger.info(f"Schemas not found for catalog {catalog.id}")
             return []
-
-        for schema in response["schemas"]:
+        for schema in response:
             yield self._create_schema(catalog, schema)
 
     def tables(self, schema: Schema) -> Iterable[Table]:
-        response: dict = self._unity_catalog_api.list_tables(
-            catalog_name=schema.catalog.name,
-            schema_name=schema.name,
-            name_pattern=None,
-        )
-
-        if response.get("tables") is None:
-            logger.info(
-                f"Tables not found for schema {schema.catalog.name}.{schema.name}"
+        with patch("databricks.sdk.service.catalog.TableInfo", TableInfoWithGeneration):
```

Comment:
Not sure if I went too far with this. This was my solution to the issue of (a) wanting to use the workspace client method and (b) still wanting to keep the `generation` field, in case we want it. That field is not documented anywhere but it seems potentially useful

Reply:
Maybe worth to ask Serge about this